draft-ietf-httpbis-compression-dictionary.md

```diff
@@ -293,24 +295,59 @@ For example:
 Content-Dictionary: :pZGm1Av0IEBKARczz7exkNYsZb8LzaMrV7J32a2fFG4=:
 ~~~
 
-# Negotiating the compression algorithm
+# Dictionary-Compressed Brotli
 
-When a compression dictionary is available for use for a given request, the
-algorithm to be used is negotiated through the regular mechanism for
-negotiating content encoding in HTTP.
+The "br-d" content encoding identifies a resource that is a
+"Dictionary-Compressed Brotli" stream.
 
-The dictionary to use is negotiated separately and advertised in the
-"Available-Dictionary" request header.
+A "Dictionary-Compressed Brotli" stream is a Brotli {{RFC7932}} stream for a
+response that has been compressed with an external dictionary using a
+compression window not larger than 16 MB.
+
+Clients that announce support for br-d content encoding MUST be able to
+decompress resources that were compressed with a window size of up to 16 MB.
+
+With Brotli compression, the full dictionary is available during compression
+and decompression independent of the compression window, allowing for
+delta-compression of resources larger than the compression window.
+
+# Dictionary-Compressed Zstandard
+
+The "zstd-d" content encoding identifies a resource that is a
+"Dictionary-Compressed Zstandard" stream.
 
-## Compression Algorithms
-This document introduces two new content encoding algorithms:
+A "Dictionary-Compressed Zstandard" stream is a Zstandard {{RFC8878}} stream
+for a response that has been compressed with an external dictionary.
 
-- br-d: Brotli {{RFC7932}} using an external compression dictionary and a compression window of not more than 16 MB.
-- zstd-d: Zstandard {{RFC8878}} using an external compression dictionary and a compression window of not more than 8 MB.
+Clients that announce support for zstd-d content encoding MUST be able to
+decompress resources that were compressed with a window size of at least 8 MB
+or 1.25 times the size of the dictionary, which ever is greater, up to a
+maximum of 128 MB.
+
+If 1.25 times the size of the dictionary results in a fractional byte window
+size, the window size should be rounded down to the nearest byte.
```

Comment:
Given that the stream will include a window size as a power of two, maybe you should just say:

```suggestion
The window size used will be encoded in the content (currently, this can be expressed
in powers of two only) and it MUST be lower than this limit. An implementation MAY
treat a window size that exceeds the limit as a decoding error.
```